src/pybind/mgr/dashboard/services/exception.py

```diff
@@ -120,3 +120,12 @@ def handle_send_command_error(component):
         yield
     except SendCommandError as e:
         raise DashboardException(e, component=component)
+
+
+@contextmanager
+def handle_orchestrator_error(component):
+    try:
+        yield
+    except RuntimeError as e:
```

Comment:
See http://docs.ceph.com/docs/master/mgr/orchestrator_modules/#error-handling

Everything that is not an `OrchestratorError` should be an internal server error. 

Every exception that is raised that should be handled gracefully should be converted to an OrchestratorError in the orchestrator modules.

Reply:
Looks like an exception on remote is always converted to RuntimeError.
https://github.com/ceph/ceph/blob/7265b55d094a639be50a567d3be92fba94c04786/src/pybind/mgr/mgr_module.py#L1269

Reply:
Yeah, I wrote that one. Within _Completions, exceptions are carefully transferred into the current subinterpreter and then re-raised.

Reply:
Thanks, I saw that. But it's weird I always catch a `RuntimeError` on an unimplemented method. I need to investigate further.